### ISSUE TITLE
how to produce the evaluation metrics numbers?

### ISSUE THREAD

**Issue by emadkavousi** [Advocate99/DiffGesture#18]
Hi and thanks for this great work . I trying to produce the metrics number of the paper but when I run the :
python scripts/test_ted.py eval
I get these logs :
python scripts/test_ted.py eval
loading checkpoint output/train_diffusion_ted/pose_diffusion_checkpoint_499.bin
epoch 499
init diffusion model
2023-12-05 10:22:03,301: PyTorch version: 1.13.1+cu117
2023-12-05 10:22:03,301: CUDA version: 11.7
2023-12-05 10:22:03,302: 1 GPUs, default cuda:0
2023-12-05 10:22:03,302: {'batch_size': 128,
 'block_depth': 8,
 'classifier_free': True,
 'config': 'config/pose_diffusion_ted.yml',
 'diff_hidden_dim': 256,
 'epochs': 500,
 'eval_net_path': 'output/train_h36m_gesture_autoencoder/gesture_autoencoder_checkpoint_best.bin',
 'freeze_wordembed': False,
 'hidden_size': 300,
 'input_context': 'audio',
 'latent_dim': 128,
 'learning_rate': 0.0005,
 'loader_workers': 4,
 'mean_dir_vec': [[0.0154009],
                  [-0.9690125],
                  [-0.0884354],
                  [-0.0022264],
                  [-0.8655276],
                  [0.4342174],
                  [-0.0035145],
                  [-0.8755367],
                  [-0.4121039],
                  [-0.9236511],
                  [0.3061306],
                  [-0.0012415],
                  [-0.5155854],
                  [0.8129665],
                  [0.0871897],
                  [0.2348464],
                  [0.1846561],
                  [0.8091402],
                  [0.9271948],
                  [0.2960011],
                  [-0.013189],
                  [0.5233978],
                  [0.8092403],
                  [0.0725451],
                  [-0.2037076],
                  [0.1924306],
                  [0.8196916]],
 'mean_pose': [[3.06e-05],
               [0.0004946],
               [0.0008437],
               [0.0033759],
               [-0.2051629],
               [-0.0143453],
               [0.0031566],
               [-0.3054764],
               [0.0411491],
               [0.0029072],
               [-0.4254303],
               [-0.001311],
               [-0.1458413],
               [-0.1505532],
               [-0.0138192],
               [-0.2835603],
               [0.0670333],
               [0.0107002],
               [-0.2280813],
               [0.112117],
               [0.2087789],
               [0.1523502],
               [-0.1521499],
               [-0.0161503],
               [0.291909],
               [0.0644232],
               [0.0040145],
               [0.2452035],
               [0.1115339],
               [0.2051307]],
 'model': 'pose_diffusion',
 'model_save_path': 'output/train_diffusion_ted',
 'motion_resampling_framerate': 15,
 'n_poses': 34,
 'n_pre_poses': 4,
 'name': 'pose_diffusion',
 'null_cond_prob': 0.1,
 'pose_dim': 27,
 'pose_representation': '3d_vec',
 'random_seed': -1,
 'save_result_video': True,
 'subdivision_stride': 10,
 'test_data_path': ['data/ted_dataset/lmdb_test'],
 'train_data_path': ['data/ted_dataset/lmdb_train'],
 'val_data_path': ['data/ted_dataset/lmdb_val'],
 'wordembed_dim': 300,
 'wordembed_path': 'data/fasttext/crawl-300d-2M-subword.bin'}
2023-12-05 10:22:03,334: Reading data 'data/ted_dataset/lmdb_val'...
2023-12-05 10:22:03,334: Found the cache data/ted_dataset/lmdb_val_cache
0

only a zero . and it dose the same for 
python scripts/test_ted.py short
I traced some part of your code and I noticed that it never enter to for loop in line of 178 of test_ted.py file . 
how can I repair your code ?

**Comment by Advocate99** [comment]
Hi, could you please check your data preparation process?
Here the 0 comes from the line 450 in test_ted.py and the line 498 in test_expressive.py.
It means you do not have the test data in the target path. So it would work after you fix the data preparation error.
Thanks.

**Comment by Ekavousi** [comment]
finally I found the solution. it was train and test cache data files. they were corrupted and because of that code could not read any data. I deleted cache folders and the code create them and its files again and now everything works fine . but now I have another question . what is the lowest loss value you could reach for test_expressive dataset during training? and thanks for reply.

**Comment by Advocate99** [comment]
> finally I found the solution. it was train and test cache data files. they were corrupted and because of that code could not read any data. I deleted cache folders and the code create them and its files again and now everything works fine . but now I have another question . what is the lowest loss value you could reach for test_expressive dataset during training? and thanks for reply.

Hi, glad to hear that. I read my logs, the lowest loss value would be 0.024-0.026.

**Comment by emadkavousi** [comment]
Hi again. Thanks for reply.I need your help in another matter. Please help me . I noticed that ted gesture and ted expressive dataset has text data too but when I trying to get text data with your dataloader I only receive an empty tensor . 
_, _, _, _, target_vec, in_audio, _, _ = data
In line 68 of train_expressive.py
I trying to get text from first argument of data but when I print it only a tensor[0] that means an empty tensor.
did you change the dataset caches and delete the text from dataset ?
please tell me how to retrieve text alongside audio from dataset. Thank you soo much.

**Comment by emadkavousi** [comment]
> Hi again. Thanks for reply.I need your help in another matter. Please help me . I noticed that ted gesture and ted expressive dataset has text data too but when I trying to get text data with your dataloader I only receive an empty tensor . _, _, _, _, target_vec, in_audio, _, _ = data In line 68 of train_expressive.py I trying to get text from first argument of data but when I print it only a tensor[0] that means an empty tensor. did you change the dataset caches and delete the text from dataset ? please tell me how to retrieve text alongside audio from dataset. Thank you soo much.

I changed 
collate_fn = default_collate_fn      line of 121 of train_expressive.py  with 
collate_fn = word_seq_collate_fn     
and now I receive a tensor of embeddings but i need raw text of each audio input. is there any way to get that ? please help me .

**Comment by Advocate99** [comment]
Hi, you can refer to the test script.
For example, in line 98-110 of test_ted.py, the raw words can be obtained.
